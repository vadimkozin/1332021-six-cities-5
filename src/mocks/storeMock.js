import configureStore from 'redux-mock-store';
import {SortingType, CITIES_NAMES, StatusType, AuthorizationStatus} from '../const';
import {offerAdapter, commentAdapter} from '../store/adapters';

import hotelJson from './json/hotel';
import hotelsNearbyJson from './json/hotels-nearby';
import offersJson from './json/offers';
import commentsJson from './json/comments';

const hotel = offerAdapter.adaptToClientOffer(hotelJson);
const hotelsNearby = offerAdapter.adaptToClient(hotelsNearbyJson);
const offers = offerAdapter.adaptToClient(offersJson);
const comments = commentAdapter.adaptToClient(commentsJson);

const initCity = CITIES_NAMES[0];

const Data = {
  SORT: `sort`,
  CITY: `city`,
  OFFERS: `offersGist`,
  HOTEL: `hotelGist`,
  HOTELS_NEARBY: `hotelsNearbyGist`,
  COMMENTS: `commentsGist`,
};

const basicState = {
  data: null,
  status: StatusType.IDLE,
  error: ``,
};

const makeInit = (basic = basicState) => (data = {}) => Object.assign({}, basic, {data});

const init = makeInit();

const storeConfig = configureStore();

const storeMock = storeConfig({
  data: {
    cities: CITIES_NAMES,
  },
  process: {
    [Data.CITY]: initCity,
    [Data.SORT]: SortingType.POPULAR,
    [Data.OFFERS]: init(offers),
    [Data.HOTEL]: init(hotel),
    [Data.HOTELS_NEARBY]: init(hotelsNearby),
    [Data.COMMENTS]: init(comments),
  },
  user: {
    id: 1,
    name: `qwerty`,
    email: `qwerty@mail.ru`,
    isPro: false,
    avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/5.jpg`,
    authorizationStatus: AuthorizationStatus.NO_AUTH,
  },
});

export {storeMock};
