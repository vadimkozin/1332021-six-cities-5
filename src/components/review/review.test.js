import React from 'react';
import {Provider} from 'react-redux';
import {render} from 'enzyme';
import storeMock, {comments} from '../../mocks/storeMock';
import Review from './review';

const review = comments[0];

describe(`Review`, () => {
  it(`should render Review component`, () => {
    const component = render(
        <Provider store={storeMock}>
          <Review review={review}/>
        </Provider>
    );

    expect(component).toMatchSnapshot();
  });
});
