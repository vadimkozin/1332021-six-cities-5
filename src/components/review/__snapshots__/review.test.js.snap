// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review should render Review component 1`] = `
<li
  class="reviews__item"
>
  <div
    class="reviews__user user"
  >
    <div
      class="reviews__avatar-wrapper user__avatar-wrapper"
    >
      <img
        alt="Reviews avatar"
        class="reviews__avatar user__avatar"
        height="54"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/9.jpg"
        width="54"
      />
    </div>
    <span
      class="reviews__user-name"
    >
      Sophie
    </span>
  </div>
  <div
    class="reviews__info"
  >
    <div
      class="reviews__rating rating"
    >
      <div
        class="reviews__stars rating__stars"
      >
        <span
          style="width:60%"
        />
        <span
          class="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <p
      class="reviews__text"
    >
      We loved it so much, the house, the veiw, the location just great.. Highly reccomend :)
    </p>
    <time
      class="reviews__time"
      datetime="2020-11-02"
    >
      November 2020
    </time>
  </div>
</li>
`;
